jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/utils/getOperands.kt

```diff
@@ -0,0 +1,215 @@
+import org.jacodb.panda.dynamic.ets.base.EtsNumberConstant
+import org.jacodb.panda.dynamic.ets.base.EtsObjectLiteral
+import org.jacodb.panda.dynamic.ets.base.EtsParameterRef
+import org.jacodb.panda.dynamic.ets.base.EtsPhiExpr
+import org.jacodb.panda.dynamic.ets.base.EtsRelationOperation
+import org.jacodb.panda.dynamic.ets.base.EtsReturnStmt
+import org.jacodb.panda.dynamic.ets.base.EtsStaticCallExpr
+import org.jacodb.panda.dynamic.ets.base.EtsStaticFieldRef
+import org.jacodb.panda.dynamic.ets.base.EtsStmt
+import org.jacodb.panda.dynamic.ets.base.EtsStringConstant
+import org.jacodb.panda.dynamic.ets.base.EtsSwitchStmt
+import org.jacodb.panda.dynamic.ets.base.EtsThis
+import org.jacodb.panda.dynamic.ets.base.EtsThrowStmt
+import org.jacodb.panda.dynamic.ets.base.EtsTypeOfExpr
+import org.jacodb.panda.dynamic.ets.base.EtsUnaryOperation
+import org.jacodb.panda.dynamic.ets.base.EtsUndefinedConstant
+
+fun EtsStmt.getOperands(): Sequence<EtsEntity> {
+    return accept(StmtGetOperands)
+}
+
+fun EtsEntity.getOperands(): Sequence<EtsEntity> {
+    return accept(EntityGetOperands)
+}
+
+private object StmtGetOperands : EtsStmt.Visitor<Sequence<EtsEntity>> {
+    override fun visit(stmt: EtsNopStmt): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(stmt: EtsAssignStmt): Sequence<EtsEntity> = sequence {
+        yield(stmt.rhv)
+    }
+
+    override fun visit(stmt: EtsCallStmt): Sequence<EtsEntity> = sequence {
+        yield(stmt.expr)
+    }
+
+    override fun visit(stmt: EtsReturnStmt): Sequence<EtsEntity> = sequence {
+        if (stmt.returnValue != null) {
+            yield(stmt.returnValue)
+        }
+    }
+
+    override fun visit(stmt: EtsThrowStmt): Sequence<EtsEntity> = sequence {
+        yield(stmt.arg)
+    }
+
+    override fun visit(stmt: EtsGotoStmt): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(stmt: EtsIfStmt): Sequence<EtsEntity> = sequence {
+        yield(stmt.condition)
+    }
+
+    override fun visit(stmt: EtsSwitchStmt): Sequence<EtsEntity> = sequence {
+        yield(stmt.arg)
+        yieldAll(stmt.cases)
+    }
+}
+
+private object EntityGetOperands : EtsEntity.Visitor<Sequence<EtsEntity>> {
+    override fun visit(value: EtsLocal): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsStringConstant): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsBooleanConstant): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsNumberConstant): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsNullConstant): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsUndefinedConstant): Sequence<EtsEntity> = sequence {
+        // empty
+    }
+
+    override fun visit(value: EtsArrayLiteral): Sequence<EtsEntity> = sequence {
```

Comment:
Why do we even need these sequences? Why not to use lists? Isn't it too expensive?

Reply:
In fact, it does not matter. The performance implications will be negligible here. However, theoretically, expressions might be very deeply nested, so I personally prefer lazy solutions (i.e., Sequence) in such cases.